### PR TITLE
Change discover snapcode aggregated output to PubSub

### PR DESCRIPTION
**Pull request by mustafasc**

### CODE REVIEW

**Comment by mustafasc** [comment]
@sandeepshresthasc

**Comment by sandeepshresthasc** [comment]
this looks good. but do you want to try out the sql based one (no code) (look at lens-stats and geofilter.properties). If you still want to run with it lgtm to me

File: aggregator/src/main/resources/discover-snapcode-stats.properties [hunk 1]
```diff
@@ -2,7 +2,6 @@ spark.app.name=discover-snapcode
 spark.streaming.window.time.period.ms=60000
 
 kafka.input.topics=blizzard.gcs.DISCOVER_SCAN_VIEW,blizzard.gcs.DISCOVER_SCAN_LINK
-output.bigquery.project=sc-analytics
-output.bigquery.datasetId=sandeep
-output.bigquery.tableId=discover_snapcode_stats_kafka
-output.bigquery.schema=scan_action_id;STRING|event_name;STRING|count;INTEGER|window_timestamp;TIMESTAMP
\ No newline at end of file
+input.projections=DISCOVER_SNAPCODE_EVENTS:event_name;StringType:scan_action_id;StringType
```
Comment (sandeepshresthasc) [inline aggregator/src/main/resources/discover-snapcode-stats.properties]: this shoud be input.projections=DISCOVER_SNAPCODE_EVENTS:event_name;StringType|scan_action_id;StringType
sorry for lack of documentation

File: aggregator/src/main/resources/discover-snapcode-stats.properties [hunk 2]
```diff
@@ -2,7 +2,6 @@ spark.app.name=discover-snapcode
...
+task.sql.discover_snapcode_stat=SELECT event_name, scan_action_id, count(*) as count from DISCOVER_SNAPCODE_EVENTS GROUP BY event_name, scan_action_id ORDER BY count DESC
```
Comment (sandeepshresthasc) [inline aggregator/src/main/resources/discover-snapcode-stats.properties]: this needs to be 2 different sql (or something like that):
task.sql.discover_snapcode_scan_view=SELECT 'DISCOVER_SCAN_VIEW', scan_action_id, count(*) AS count FROM DISCOVER_SCAN_VIEW GROUP BY scan_action_id ORDER BY count DESC
task.sql.discover_snapcode_scan_link=SELECT 'DISCOVER_SCAN_LINK', scan_action_id, count(*) AS count FROM DISCOVER_SCAN_LINK GROUP BY scan_action_id ORDER BY count DESC

File: aggregator/src/main/resources/discover-snapcode-stats.properties [hunk 3]
```diff
@@ -2,7 +2,6 @@ spark.app.name=discover-snapcode
...
+output.pubsub.topic=projects/sc-analytics/topics/realtime-output-discover-snapcode
```
Comment (sandeepshresthasc) [inline aggregator/src/main/resources/discover-snapcode-stats.properties]: this needs to be 2 output task to match the 2 sql task above:
output.pubsub.discover_snapcode_scan_view=projects/sc-analytics/topics/realtime-output-discover-snapcode
output.pubsub.discover_snapcode_scan_link=projects/sc-analytics/topics/realtime-output-discover-snapcode

File: aggregator/src/main/resources/discover-snapcode-stats.properties (hunk 1 above)
Comment (sandeepshresthasc) [inline aggregator/src/main/resources/discover-snapcode-stats.properties]: this needs projection for both tables (event_name/topic)
input.projections=DISCOVER_SCAN_VIEW:scan_action_id;StringType,DISCOVER_SCAN_LINK:scan_action_id;StringType

**Comment by mustafasc** [comment]
Updated to output data into separate streams. Set to BigQuery temporarily to get approval from Broadcast team before switching to PubSub.

If you wouldn't mind also creating the temp BigQuery tables. Thanks!

**Comment by sandeepshresthasc** [comment]
lgtm

**Comment by mustafasc** [comment]
Squash merge commit f3b605e4b526ea184ac4af768d105d9b83fe77ef created by [Whaler](https://chrome.google.com/webstore/detail/whaler/kjfaikbmbbkbnjjeiambmjchclpfkedf). :whale:

**Pull request closed without merging**